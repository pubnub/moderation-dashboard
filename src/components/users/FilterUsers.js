import React from "react";
import { useStyles } from "../../style/drawer";
import Drawer from "@material-ui/core/Drawer";
import Button from "@material-ui/core/Button";
import {
  Avatar,
  Box,
  Checkbox,
  Grid,
  IconButton,
  List,
  ListItem,
  ListItemIcon,
  ListItemText,
  Typography,
} from "@material-ui/core";

export default function FilterUsers(props) {
  const classes = useStyles();
  const [open, setOpen] = React.useState(false);
  const [isBanChecked, setIsBanChecked] = React.useState(false);
  const [isFlagChecked, setIsFlagChecked] = React.useState(false);

  const handleClickOpen = () => {
    setOpen(true);
  };

  const handleClose = () => {
    setOpen(false);
  };

  const handleChange = (action) => {
    if (action === "ban" && !isBanChecked) {
      setIsFlagChecked(false);
      setIsBanChecked(true);
      props.isFiltered(action);
    } else if (action === "flag" && !isFlagChecked) {
      setIsBanChecked(false);
      setIsFlagChecked(true);
      props.isFiltered(action);
    } else {
      props.isFiltered("all");
      setIsBanChecked(false);
      setIsFlagChecked(false);
    }
  };

  return (
    <>
      <Avatar id="openDrawer" variant="square" onClick={handleClickOpen} className={classes.avatar}>
        <IconButton>
          <img src={process.env.PUBLIC_URL + "/images/filter.svg"} alt="filter" />
        </IconButton>
      </Avatar>
      <Drawer anchor="right" open={open} onClose={handleClose}>
        <div className={classes.list}>
          <Box pl={3} pt={4} pr={1}>
            <Grid justify="space-between" container>
              <Grid item>
                <Typography className={classes.title}>Filters</Typography>
              </Grid>
              <Grid item>
                <Button id="drawer" onClick={handleClose}>
                  <img src={process.env.PUBLIC_URL + "/images/times.svg"} alt="filter" />
                </Button>
              </Grid>
            </Grid>
          </Box>
          <Box pl={3} pt={4}>
            <Typography testid="UserType" className={classes.userType}>
              User Type
            </Typography>
          </Box>
          <List>
            <ListItem button onClick={() => handleChange("ban")} testid="banButton">
              <ListItemIcon>
                <Checkbox
                  id="banCheck"
                  name="banned_users"
                  value="ban"
                  color="primary"
                  checked={isBanChecked}
                />
              </ListItemIcon>
              <ListItemText>
                <Typography testid="bannedUsers" className={classes.listText}>
                  Banned Users
                </Typography>
              </ListItemText>
            </ListItem>
            <ListItem button onClick={() => handleChange("flag")} testid="flagButton">
              <ListItemIcon>
                <Checkbox
                  id="flagCheck"
                  name="flagged_users"
                  value="flag"
                  color="primary"
                  checked={isFlagChecked}
                />
              </ListItemIcon>
              <ListItemText>
                <Typography testid="flaggedUsers" className={classes.listText}>
                  Flagged Users
                </Typography>
              </ListItemText>
            </ListItem>
          </List>
        </div>
      </Drawer>
    </>
  );
}
