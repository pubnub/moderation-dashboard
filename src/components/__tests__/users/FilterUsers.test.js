import React from "react";
import { shallow } from "enzyme";
import FilterUsers from "../../users/FilterUsers";

describe("Filter User meta data Modal", () => {
  let component;
  beforeAll(() => {
    component = shallow(<FilterUsers isFiltered={jest.fn()} />);
  });

  test("Snapshhot", () => {
    expect(component).toMatchSnapshot();
  });

  test("Has Label", () => {
    expect(component.find('[testid="UserType"]').text()).toBe("User Type");
    expect(component.find('[testid="bannedUsers"]').text()).toBe("Banned Users");
    expect(component.find('[testid="flaggedUsers"]').text()).toBe("Flagged Users");
  });

  test("initial Values", () => {
    expect(component.find("#banCheck").getElement().props.checked).toBeFalsy();
    expect(component.find("#flagCheck").getElement().props.checked).toBeFalsy();
    expect(component.find('input[type="checkbox"][checked="checked"]')).toHaveLength(0);
  });

  test("handle Drawer", () => {
    const drawerOpen = component.find("#openDrawer").simulate("click");
    expect(drawerOpen).toBeTruthy();

    const drawerClose = component.find("#drawer").simulate("click");
    expect(drawerClose).toBeTruthy();
  });

  test("check handleChange Function", () => {
    component.find('[testid="banButton"]').simulate("click");
    expect(component.find("#banCheck").getElement().props.checked).toBe(true);
    expect(component.find("#flagCheck").getElement().props.checked).toBe(false);

    component.find('[testid="flagButton"]').simulate("click");
    expect(component.find("#flagCheck").getElement().props.checked).toBe(true);
    expect(component.find("#banCheck").getElement().props.checked).toBe(false);

    component.find('[testid="flagButton"]').simulate("click");
    expect(component.find("#flagCheck").getElement().props.checked).toBe(false);
    expect(component.find("#banCheck").getElement().props.checked).toBe(false);
  });
});
