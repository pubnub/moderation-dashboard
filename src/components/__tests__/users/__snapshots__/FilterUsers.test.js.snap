// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter User meta data Modal Snapshhot 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Avatar))
    className="makeStyles-avatar-3"
    id="openDrawer"
    onClick={[Function]}
    variant="square"
  >
    <WithStyles(ForwardRef(IconButton))>
      <img
        alt="filter"
        src="/images/filter.svg"
      />
    </WithStyles(ForwardRef(IconButton))>
  </WithStyles(ForwardRef(Avatar))>
  <WithStyles(ForwardRef(Drawer))
    anchor="right"
    onClose={[Function]}
    open={false}
  >
    <div
      className="makeStyles-list-1"
    >
      <Styled(MuiBox)
        pl={3}
        pr={1}
        pt={4}
      >
        <WithStyles(ForwardRef(Grid))
          container={true}
          justify="space-between"
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
          >
            <WithStyles(ForwardRef(Typography))
              className="makeStyles-title-2"
            >
              Filters
            </WithStyles(ForwardRef(Typography))>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
          >
            <WithStyles(ForwardRef(Button))
              id="drawer"
              onClick={[Function]}
            >
              <img
                alt="filter"
                src="/images/times.svg"
              />
            </WithStyles(ForwardRef(Button))>
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
      </Styled(MuiBox)>
      <Styled(MuiBox)
        pl={3}
        pt={4}
      >
        <WithStyles(ForwardRef(Typography))
          className="makeStyles-userType-5"
          testid="UserType"
        >
          User Type
        </WithStyles(ForwardRef(Typography))>
      </Styled(MuiBox)>
      <WithStyles(ForwardRef(List))>
        <WithStyles(ForwardRef(ListItem))
          button={true}
          onClick={[Function]}
          testid="banButton"
        >
          <WithStyles(ForwardRef(ListItemIcon))>
            <WithStyles(ForwardRef(Checkbox))
              checked={false}
              color="primary"
              id="banCheck"
              name="banned_users"
              value="ban"
            />
          </WithStyles(ForwardRef(ListItemIcon))>
          <WithStyles(ForwardRef(ListItemText))>
            <WithStyles(ForwardRef(Typography))
              className="makeStyles-listText-4"
              testid="bannedUsers"
            >
              Banned Users
            </WithStyles(ForwardRef(Typography))>
          </WithStyles(ForwardRef(ListItemText))>
        </WithStyles(ForwardRef(ListItem))>
        <WithStyles(ForwardRef(ListItem))
          button={true}
          onClick={[Function]}
          testid="flagButton"
        >
          <WithStyles(ForwardRef(ListItemIcon))>
            <WithStyles(ForwardRef(Checkbox))
              checked={false}
              color="primary"
              id="flagCheck"
              name="flagged_users"
              value="flag"
            />
          </WithStyles(ForwardRef(ListItemIcon))>
          <WithStyles(ForwardRef(ListItemText))>
            <WithStyles(ForwardRef(Typography))
              className="makeStyles-listText-4"
              testid="flaggedUsers"
            >
              Flagged Users
            </WithStyles(ForwardRef(Typography))>
          </WithStyles(ForwardRef(ListItemText))>
        </WithStyles(ForwardRef(ListItem))>
      </WithStyles(ForwardRef(List))>
    </div>
  </WithStyles(ForwardRef(Drawer))>
</Fragment>
`;
